writer.js

```diff
@@ -0,0 +1,85 @@
+const nsqStream = require('nsq-stream');
+
+/**
+ * @constructor
+ * @param {Object} opts - options for the writer
+ */
+function Writer(opts = {}) {
+  const { writer, topic, spec = {}, log } = opts;
+  this.log = log;
+  this.writeStream = writer && topic && nsqStream.createWriteStream(writer, topic);
+  this.spec = spec;
+}
+
+/**
+ * Combines the spec with status information
+ *
+ * @function buildStatusMessage
+ * @param {Object} statusInfo - Status information
+ * @returns {Object} - The combined message object
+ * @api private
+ */
+Writer.prototype.buildStatusMessage = function buildStatusMessage(statusInfo) {
+  const { type: buildType, ...other } = this.spec;
+
+  if (statusInfo.error) {
+    statusInfo.eventType = 'error';
+    statusInfo.message = statusInfo.error.message;
```

Comment:
we actually want to also look for a special `error.output` property that comes back from `workers-factory` which gives us the FULL web pack error. This will be the essential one. Reference https://github.com/warehouseai/workers-factory/blob/master/workers/webpack.js#L129. We also output it explicitly in the error logs in the stream handler.